scipy/optimize/lbfgsb.py

```diff
@@ -250,6 +251,10 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
         bounds = [(None, None)] * n
     if len(bounds) != n:
         raise ValueError('length of x0 != length of bounds')
+    # unbounded variables must use None, not +-inf, for optimizer to work properly
+    l = [None if l == -np.inf else l for l, _ in bounds]
+    u = [None if u == np.inf else u for _, u in bounds]
+    bounds = zip(l, u)
```

Comment:
In python3, `zip` creates a "zip" object, which is not subscriptable as is a list (which `zip` returns in Python2). So here, you could do `bounds = list(zip(l, u))` to keep a list, as subscriptability is expected later on, or do this in a single list comprehension without a zip.